Subject: [PATCH] feat: modify config
---
Index: Telegram/SourceFiles/config.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/config.h b/Telegram/SourceFiles/config.h
--- a/Telegram/SourceFiles/config.h	(revision 57c01bbfb18ce0a63fe86d7e998f4a9ad7356915)
+++ b/Telegram/SourceFiles/config.h	(date 1684019758656)
@@ -21,9 +21,9 @@
 
 	AutoSearchTimeout = 900, // 0.9 secs
 
-	PreloadHeightsCount = 3, // when 3 screens to scroll left make a preload request
+	PreloadHeightsCount = 4, // when 4 screens to scroll left make a preload request
 
-	SearchPeopleLimit = 5,
+	SearchPeopleLimit = 20,
 
 	MaxMessageSize = 4096,
 
