Subject: [PATCH] feat: use AyuGram branding where possible
---
Index: Telegram/SourceFiles/data/data_session.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/data/data_session.cpp b/Telegram/SourceFiles/data/data_session.cpp
--- a/Telegram/SourceFiles/data/data_session.cpp	(revision 57c01bbfb18ce0a63fe86d7e998f4a9ad7356915)
+++ b/Telegram/SourceFiles/data/data_session.cpp	(date 1683129316903)
@@ -4180,7 +4180,7 @@
 				| MTPDuser::Flag::f_verified),
 			MTP_long(peerToUser(PeerData::kServiceNotificationsId).bare),
 			MTPlong(), // access_hash
-			MTP_string("Telegram"),
+			MTP_string("AyuGram"),
 			MTPstring(), // last_name
 			MTPstring(), // username
 			MTP_string("42777"),
Index: Telegram/SourceFiles/window/window_main_menu.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/window/window_main_menu.cpp b/Telegram/SourceFiles/window/window_main_menu.cpp
--- a/Telegram/SourceFiles/window/window_main_menu.cpp	(revision 57c01bbfb18ce0a63fe86d7e998f4a9ad7356915)
+++ b/Telegram/SourceFiles/window/window_main_menu.cpp	(date 1683129392446)
@@ -459,8 +459,8 @@
 	parentResized();
 
 	_telegram->setMarkedText(Ui::Text::Link(
-		u"64Gram Desktop"_q,
-		u"https://github.com/TDesktop-x64/tdesktop"_q));
+            u"AyuGram Desktop"_q,
+            u"https://radolyn.com"_q));
 	_telegram->setLinksTrusted();
 	_version->setMarkedText(
 		Ui::Text::Link(
