Subject: [PATCH] feat: show peer id
---
Index: Telegram/SourceFiles/info/profile/info_profile_actions.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/info/profile/info_profile_actions.cpp b/Telegram/SourceFiles/info/profile/info_profile_actions.cpp
--- a/Telegram/SourceFiles/info/profile/info_profile_actions.cpp	(revision a6e5e7ab84a4c89fda3e571f5d3322cf0aeac306)
+++ b/Telegram/SourceFiles/info/profile/info_profile_actions.cpp	(date 1681410735723)
@@ -479,6 +479,12 @@
 			});
 		}
 
+        // AyuGram: show id
+        addInfoOneLine(
+                rpl::single(QString("ID")),
+                IDValue(user),
+                QString("Copy ID"));
+
 		AddMainButton(
 			result,
 			tr::lng_info_add_as_contact(),
Index: Telegram/SourceFiles/info/profile/info_profile_values.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/info/profile/info_profile_values.cpp b/Telegram/SourceFiles/info/profile/info_profile_values.cpp
--- a/Telegram/SourceFiles/info/profile/info_profile_values.cpp	(revision a6e5e7ab84a4c89fda3e571f5d3322cf0aeac306)
+++ b/Telegram/SourceFiles/info/profile/info_profile_values.cpp	(date 1681410808381)
@@ -87,6 +87,11 @@
 
 } // namespace
 
+// AyuGram: show id (thx to Kotatogram)
+rpl::producer<TextWithEntities> IDValue(not_null<PeerData*> peer) {
+    return rpl::single(QString::number(peer->id.value)) | Ui::Text::ToWithEntities();
+}
+
 rpl::producer<QString> NameValue(not_null<PeerData*> peer) {
 	return peer->session().changes().peerFlagsValue(
 		peer,
Index: Telegram/SourceFiles/info/profile/info_profile_values.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/info/profile/info_profile_values.h b/Telegram/SourceFiles/info/profile/info_profile_values.h
--- a/Telegram/SourceFiles/info/profile/info_profile_values.h	(revision a6e5e7ab84a4c89fda3e571f5d3322cf0aeac306)
+++ b/Telegram/SourceFiles/info/profile/info_profile_values.h	(date 1681409545476)
@@ -44,6 +44,7 @@
 rpl::producer<not_null<PeerData*>> MigratedOrMeValue(
 	not_null<PeerData*> peer);
 
+[[nodiscard]] rpl::producer<TextWithEntities> IDValue(not_null<PeerData*> peer);
 [[nodiscard]] rpl::producer<QString> NameValue(not_null<PeerData*> peer);
 [[nodiscard]] rpl::producer<QString> TitleValue(
 	not_null<Data::ForumTopic*> topic);
