Subject: [PATCH] feat: implement sendOnlinePackets & sendOfflinePacketAfterOnline
---
Index: Telegram/SourceFiles/api/api_updates.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/api/api_updates.cpp b/Telegram/SourceFiles/api/api_updates.cpp
--- a/Telegram/SourceFiles/api/api_updates.cpp	(revision a6e5e7ab84a4c89fda3e571f5d3322cf0aeac306)
+++ b/Telegram/SourceFiles/api/api_updates.cpp	(date 1680899622368)
@@ -899,8 +899,17 @@
 		Core::App().checkAutoLock(lastNonIdleTime);
 	});
 
+    // AyuGram sendOnlinePackets
+    const auto settings = &Core::App().settings();
 	const auto &config = _session->serverConfig();
-	bool isOnline = Core::App().hasActiveWindow(&session());
+    bool isOnlineOrig = Core::App().hasActiveWindow(&session());
+    bool isOnline = settings->sendOnlinePackets() && isOnlineOrig;
+
+    // AyuGram sendOfflinePacketAfterOnline
+    if (settings->sendOfflinePacketAfterOnline() && _lastWasOnline) {
+        isOnline = false;
+    }
+
 	int updateIn = config.onlineUpdatePeriod;
 	Assert(updateIn >= 0);
 	if (isOnline) {
@@ -922,7 +931,7 @@
 		|| (isOnline && gotOtherOffline)) {
 		api().request(base::take(_onlineRequest)).cancel();
 
-		_lastWasOnline = isOnline;
+		_lastWasOnline = isOnlineOrig;
 		_lastSetOnline = ms;
 		if (!Core::Quitting()) {
 			_onlineRequest = api().request(MTPaccount_UpdateStatus(
@@ -943,7 +952,7 @@
 		session().changes().peerUpdated(
 			self,
 			Data::PeerUpdate::Flag::OnlineStatus);
-		if (!isOnline) { // Went offline, so we need to save message draft to the cloud.
+		if (!isOnlineOrig) { // Went offline, so we need to save message draft to the cloud.
 			api().saveCurrentDraftToCloud();
 			session().data().maybeStopWatchForOffline(self);
 		}
@@ -953,6 +962,22 @@
 		updateIn = qMin(updateIn, int(_lastSetOnline + config.onlineUpdatePeriod - ms));
 		Assert(updateIn >= 0);
 	}
+
+    // AyuGram sendOfflinePacketAfterOnline
+    if (settings->sendOfflinePacketAfterOnline()) {
+        session().api().requestFullPeer(session().user());
+        if (session().user()->onlineTill > base::unixtime::now()) {
+            DEBUG_LOG(("[AyuGram] User likely appeared online"));
+
+            _onlineRequest = api().request(MTPaccount_UpdateStatus(
+                    MTP_bool(true)
+            )).send();
+        }
+
+        DEBUG_LOG(("[AyuGram] Decreasing updateIn because of enabled features"));
+        updateIn = 1250;
+    }
+
 	_onlineTimer.callOnce(updateIn);
 }
 
