Subject: [PATCH] feat: implement read buttons
---
Index: Telegram/SourceFiles/window/window_peer_menu.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/window/window_peer_menu.cpp b/Telegram/SourceFiles/window/window_peer_menu.cpp
--- a/Telegram/SourceFiles/window/window_peer_menu.cpp	(revision 57c01bbfb18ce0a63fe86d7e998f4a9ad7356915)
+++ b/Telegram/SourceFiles/window/window_peer_menu.cpp	(date 1683717827764)
@@ -1369,6 +1369,11 @@
 
 } // namespace
 
+// I hate C++ actually
+void MarkAsReadChatListHack(not_null<Dialogs::MainList*> list) {
+    MarkAsReadChatList(list);
+}
+
 void PeerMenuHidePinnedMessage(not_null<PeerData*> peer) {
 	auto hidden = HistoryWidget::switchPinnedHidden(peer, true);
 	if (hidden) {
Index: Telegram/SourceFiles/window/window_peer_menu.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/window/window_peer_menu.h b/Telegram/SourceFiles/window/window_peer_menu.h
--- a/Telegram/SourceFiles/window/window_peer_menu.h	(revision 57c01bbfb18ce0a63fe86d7e998f4a9ad7356915)
+++ b/Telegram/SourceFiles/window/window_peer_menu.h	(date 1683717801064)
@@ -177,4 +177,6 @@
 
 void AddSeparatorAndShiftUp(const PeerMenuCallback &addAction);
 
+void MarkAsReadChatListHack(not_null<Dialogs::MainList*> list);
+
 } // namespace Window
Index: Telegram/SourceFiles/window/window_main_menu.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Telegram/SourceFiles/window/window_main_menu.cpp b/Telegram/SourceFiles/window/window_main_menu.cpp
--- a/Telegram/SourceFiles/window/window_main_menu.cpp	(revision 57c01bbfb18ce0a63fe86d7e998f4a9ad7356915)
+++ b/Telegram/SourceFiles/window/window_main_menu.cpp	(date 1683718691949)
@@ -76,6 +76,12 @@
 #include <QtGui/QGuiApplication>
 #include <QtGui/QClipboard>
 
+
+// AyuGram includes
+#include "history/history.h"
+#include "data/data_histories.h"
+
+
 namespace Window {
 namespace {
 
@@ -459,8 +465,8 @@
 	parentResized();
 
 	_telegram->setMarkedText(Ui::Text::Link(
-		u"64Gram Desktop"_q,
-		u"https://github.com/TDesktop-x64/tdesktop"_q));
+            u"AyuGram Desktop"_q,
+            u"https://radolyn.com"_q));
 	_telegram->setLinksTrusted();
 	_version->setMarkedText(
 		Ui::Text::Link(
@@ -766,6 +772,32 @@
 		)->setClickedCallback([=] {
 			controller->showPeerHistory(controller->session().user());
 		});
+        addAction(
+                rpl::single(QString("LRead Messages")),
+                { &st::settingsIconForward, kIconPurple }
+        )->setClickedCallback([=] {
+            const auto settings = &Core::App().settings();
+            const auto prev = settings->sendReadPackets();
+            settings->setSendReadPackets(false);
+
+            auto chats = controller->session().data().chatsList();
+            MarkAsReadChatListHack(chats);
+
+            settings->setSendReadPackets(prev);
+        });
+        addAction(
+                rpl::single(QString("SRead Messages")),
+                { &st::settingsIconForward, kIconPurple }
+        )->setClickedCallback([=] {
+            const auto settings = &Core::App().settings();
+            const auto prev = settings->sendReadPackets();
+            settings->setSendReadPackets(true);
+
+            auto chats = controller->session().data().chatsList();
+            MarkAsReadChatListHack(chats);
+
+            settings->setSendReadPackets(prev);
+        });
 	} else {
 		addAction(
 			tr::lng_profile_add_contact(),
